exporter/clickhouseexporter/config.go

```diff
@@ -30,18 +33,14 @@ type Config struct {
 	// because only QueueSize is user-settable.
 	QueueSettings QueueSettings `mapstructure:"sending_queue"`
 
-	// Endpoint is the clickhouse server endpoint.
-	// TCP endpoint: tcp://ip1:port,ip2:port
-	// HTTP endpoint: http://ip:port,ip2:port
+	// Endpoint is the clickhouse endpoint.
 	Endpoint string `mapstructure:"endpoint"`
+	// Database is the database name to export.
+	Database string `mapstructure:"database"`
 	// Username is the authentication username.
 	Username string `mapstructure:"username"`
 	// Username is the authentication password.
 	Password string `mapstructure:"password"`
-	// Database is the database name to export.
-	Database string `mapstructure:"database"`
-	// ConnectionParams is the extra connection parameters with map format. for example compression/dial_timeout
-	ConnectionParams map[string]string `mapstructure:"connection_params"`
```

Comment:
Why do we need to move the params from this option to the endpoint if we parse them after that anyway? It's also a breaking change which is ok for the alpha component but still better to avoid if there are no strong reasons for that

Reply:
The most common approach is to set query parameters on the [DSN](https://github.com/ClickHouse/clickhouse-go#dsn). We could have both, but it would have made it a bit more complicated. I'd like to use the endpoint-only approach.

Reply:
I agree it should be one explicit way to configure the DSN. But it's inconsistent now. We ask for Database, Username, and Password separately, but not ConnectionParams.

Reply:
It's an improvement. I removed `ConnectionParams`. I kept `Username` and `Password` separate because it's handy when using a secret store. Some people store the full DSN and some only the credentials.

Reply:
We could also just go with the endpoint if you don't agree.

Reply:
Restored.

Reply:
I restored the `ConnectionParams` option. The PR only contains the TLS fix with some internal improvements in how DSN is handled. Tests were added. Readme changes and dep upgrades are minor things. If you still think this is not mergeable, please help; otherwise, I will close the PR.